bull/pdl-queue/consumer.js

```diff
@@ -112,27 +140,78 @@ async function uploadToIA(zip, metadata, byteLength, email, job) {
   );
 }
 
+async function uploadPdfToIA(pdf, metadata, byteLength, email, job) {
+  const bucketTitle = metadata.IAIdentifier;
+  const IAuri = `http://s3.us.archive.org/${bucketTitle}/${bucketTitle}.pdf`;
+
+  let headers = setHeaders(
+    metadata,
+    byteLength,
+    metadata.title,
+    job.data.details.contentType
+  );
+  const options = {
+    method: "PUT",
+    uri: IAuri,
+    headers: headers,
+  };
+  const response = await new Promise((resolve, reject) => {
```

Comment:
@okerekechinweotito Please check the code in `consumer.js` for Google Books Queue and try to follow the same process as that is heavily tested on scale.

Reply:
fixed